File: docflow-frontend/types/api/AuditExportVerificationResultDTO.ts
export type AuditExportVerificationResultDTO = {
  ok: boolean;

  snapshotId: string; // UUID
  stream: string | null;
  tenantId: string | null;

  exportFrom: string | null; // ISO Instant
  exportTo: string | null;

  expectedRowCount: number;

  expectedPayloadSha256Hex: string | null;
  computedPayloadSha256Hex: string | null;

  digestMatches: boolean;

  keyIdExists: boolean;
  signatureValid: boolean;

  signatureAlgorithm: string | null;
  keyId: string | null;

  metaPresent: boolean;
  metaParsed: boolean;
  metaSnapshotIdMatches: boolean;
  metaDigestMatches: boolean;
  metaSignatureMatches: boolean;
  metaKeyIdMatches: boolean;
  metaAlgorithmMatches: boolean;
  metaDigestAlgorithmMatches: boolean;
  metaSignatureInputMatches: boolean;
  metaStreamMatches: boolean;
  metaRangeMatches: boolean;
  metaTenantMatches: boolean;
  metaRowCountMatches: boolean;

  message: string;
};

File: docflow-backend/src/main/java/com/brutecx/docflow_backend/api/dto/audit/AuditExportVerificationResultDTO.java
package com.brutecx.docflow_backend.api.dto.audit;

import java.time.Instant;
import java.util.UUID;

public record AuditExportVerificationResultDTO(
        boolean ok,
        UUID snapshotId,
        String stream,
        UUID tenantId,
        Instant exportFrom,
        Instant exportTo,
        long expectedRowCount,
        String expectedPayloadSha256Hex,
        String computedPayloadSha256Hex,
        boolean digestMatches,
        boolean keyIdExists,
        boolean signatureValid,
        String signatureAlgorithm,
        String keyId,
        boolean metaPresent,
        boolean metaParsed,
        boolean metaSnapshotIdMatches,
        boolean metaDigestMatches,
        boolean metaSignatureMatches,
        boolean metaKeyIdMatches,
        boolean metaAlgorithmMatches,
        boolean metaDigestAlgorithmMatches,
        boolean metaSignatureInputMatches,
        boolean metaStreamMatches,
        boolean metaRangeMatches,
        boolean metaTenantMatches,
        boolean metaRowCountMatches,
        String message
) {
}
